mysql-test/mysql-test-run.pl

```diff
@@ -6155,7 +6155,7 @@ ($)
                         $opt_max_save_datadir, set to 0 for no limit. Set
                         it's default with MTR_MAX_SAVE_DATADIR
   max-test-fail         Limit the number of test failures before aborting
-                        the current test run. Defaults to
+                        the current test run. Requires --force. Defaults to
```

Comment:
I'd prefer a straight fix:
```diff
--- a/mysql-test/mysql-test-run.pl
+++ b/mysql-test/mysql-test-run.pl
@@ -1396,6 +1396,11 @@ sub command_line_setup {
     $opt_big_test= 1;
   }
 
+  if ( !$opt_max_test_fail && !$opt_force )
+  {
+    $opt_force= 1;
+  }
+
   # --------------------------------------------------------------------------
   # Find out type of logging that are being used
   # --------------------------------------------------------------------------
@@ -6342,7 +6347,7 @@ Options for debugging the product
                         $opt_max_save_datadir, set to 0 for no limit. Set
                         it's default with MTR_MAX_SAVE_DATADIR
   max-test-fail         Limit the number of test failures before aborting
-                        the current test run. Defaults to
+                        the current test run. Implies --force. Defaults to
                         $opt_max_test_fail, set to 0 for no limit. Set
                         it's default with MTR_MAX_TEST_FAIL
   core-in-failure      Generate a core even if run server is run with valgrind
```

Reply:
Then it makes --force redundant. Maybe delete it completely? Or mark deprecated at least

Reply:
https://github.com/MariaDB/server/pull/1332/commits/fffefd3ce1508e0abc7d01bd59d4b56cb6828dbf

Reply:
So we can treat this patch (fffefd3) as a special case for all tests (`=0`) and is covered by documentation (in patch). 
However `--force` has to be applied while specifying number of failed tests `./mtr --suite=anel --max-test-fail=2 --force` which I see as non-unique rule. At least documentation has to say about this case.

Reply:
Actually `$opt_max_test_fail != 1` is a better condition.

Reply:
> Then it makes --force redundant. Maybe delete it completely? Or mark deprecated at least
I don't know why no `--force` is needed neither. Seems like implementation conflict (that who did second option was unaware of first).

Reply:
https://github.com/MariaDB/server/commit/b2c5019c8b25983cd896b74a62bfdced6c0a578c :-)

Reply:
I would be happy. But maybe @montywi  would not...